### ISSUE TITLE
Regex of RelabelConfig not work right

### ISSUE THREAD

**Issue by zgfh** [VictoriaMetrics/VictoriaMetrics#2928]
**Describe the bug**

```
- regex: xx_(foo|bar)
```
wrong Marshal to 
```
- regex:\n  - xx_(foo\n  - bar\n
```

**To Reproduce**
Steps to reproduce the behavior.

when create this cr:
```
apiVersion: operator.victoriametrics.com/v1beta1
kind: VMServiceScrape
metadata:
  annotations:
  labels:
    app.kubernetes.io/name: kubelet
    app.kubernetes.io/part-of: kube-prometheus
  name: kubelet
  namespace: monitoring
spec:
  endpoints:
  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
    honorLabels: true
    honorTimestamps: false
    interval: 30s
    metricRelabelConfigs:
    - action: drop
      regex: container_(network_tcp_usage_total|network_udp_usage_total|tasks_state|cpu_load_average_10s)
      sourceLabels:
      - __name__
    - action: drop
      regex: (container_spec_.*|container_file_descriptors|container_sockets|container_threads_max|container_threads|container_start_time_seconds|container_last_seen);;
      sourceLabels:
      - __name__
      - pod
      - namespace
    - action: drop
      regex: (container_blkio_device_usage_total);.+
      sourceLabels:
      - __name__
      - container
    path: /metrics/cadvisor
    port: https-metrics
    relabelConfigs:
    - action: replace
      sourceLabels:
      - __metrics_path__
      targetLabel: metrics_path
    scheme: https
    tlsConfig:
      ca: {}
      cert: {}
      insecureSkipVerify: true
  

```
then view vmagent ui: http://127.0.0.1:8429/config
find wong config:

```
...
  metric_relabel_configs:
  - source_labels: [__name__]
    regex:
    - container_(network_tcp_usage_total
    - network_udp_usage_total
    - tasks_state
    - cpu_load_average_10s)
    action: drop
  - source_labels: [__name__, pod, namespace]
    regex:
    - (container_spec_.*
    - container_file_descriptors
    - container_sockets
    - container_threads_max
    - container_threads
    - container_start_time_seconds
    - container_last_seen);;
    action: drop
  - source_labels: [__name__, container]
    regex: (container_blkio_device_usage_total);.+
    action: drop
...

```

**Expected behavior**

**Version**
v1.79.0 (The master branch is also not fixed)

**Comment by remones** [comment]
Prometheus uses the "regex" field as the origin input argument of `regexp.MatchString`， VM split the regex string by `|`，and match it one by one

**Comment by valyala** [comment]
@remones , thanks for the bugreport! The bug should be fixed in the commit 68c4ec9472bba3011c7f70a526a4ae6d99410be5 . This commit will be included in the next release. In the meantime you can build `vmagent` from this commit according to [these docs](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources) and check whether the issue is fixed there.

**Comment by valyala** [comment]
`vmagent` shouldn't split `regex` into multiple lines if it contains groups in `(...)` and `[...]` braces starting from [v1.79.2](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). This should fix the original bug. Closing the issue as fixed.

@remones , feel free re-opening the issue if the bug still exists in [vmagent v1.79.2](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
